### ISSUE TITLE
[FEATURE]: Manual tiling (i.e. windows-style snapping)

### ISSUE THREAD

**Issue by ghost** [Bismuth-Forge/bismuth#108]
### Summary

Manual tiling like windows, with joing resizing. This is something nearly every desktop has nowadays, except for KDE.

### A clear and concise description of what you want to happen

Snap window to one side, resize it.
Snap another window to the other side, have it fill the empty space.
Resize any tiled window, and have the other window resized automatically in tandem.
Un-snapping a window returns the original floating size.

Basically everything that windows does, minus the snap assist which would be nice but is also a lot of work.  It would require drawing an overview like https://github.com/tcorreabr/Parachute

### A clear and concise description of any alternative solutions or features you've considered.

there are other kwin scripts that try to do this but they are unmaintained and bugged.
